### DISCUSSION TITLE
Assessment of the difficulty in porting CPU architecture for libgit2

### DISCUSSION THREAD

**Discussion by wangyuliu** [libgit2/libgit2#6635] [Q&A] [answered]
Hello everyone! I am working on implementing a tool to assess the complexity of CPU architecture porting. It primarily focuses on RISC-V architecture porting. In fact, the tool may have an average estimate of various architecture porting efforts.My focus is on the overall workload and difficulty of transplantation in the past and future,even if a project has already been ported.As part of my dataset, I have collected the **libgit2** project. **I would like to gather community opinions to support my assessment. I appreciate your help and response!** Based on scanning tools, the porting complexity is determined to be simple, with a simple amount of code related to the CPU architecture in the project.  Is this assessment accurate? I look forward to your help and response.

**Comment by ethomson** [comment, marked as answer]
I believe that yes, it should be relatively straightforward. We ported to ARM, for instance, with minimal changes. (IIRC, the only one was around the default sign for chars.)

I have not done a build for RISC but we have no assembly in the library itself (possibly in a third party dependency like zlib) and few assumptions about the underlying architecture that should be well-ifdef'd.
